lib/Core/Inference/FrameBuilder/AbstractInstanceOfWalker.php

```diff
@@ -67,24 +83,73 @@ protected function variableFromBinaryExpression(BinaryExpression $node)
 
         $type = (string) $rightOperand->getResolvedName();
 
-        $context = $this->createSymbolContext($variable);
+        $context = $this->createSymbolContext($variable, $frame);
         $context = $context->withType(Type::fromString($type));
         $variable = WorseVariable::fromSymbolContext($context);
 
         return $variable;
     }
 
-    protected function createSymbolContext(Variable $leftOperand)
+    protected function createSymbolContext(Expression $leftOperand, Frame $frame): SymbolContext
     {
-        $context = $this->symbolFactory()->context(
-            $leftOperand->getName(),
+        assert($leftOperand instanceof Variable || $leftOperand instanceof MemberAccessExpression);
+
+        if ($leftOperand instanceof MemberAccessExpression) {
+            return $this->createPropertySymbolContext($leftOperand, $frame);
+        }
+
+        return $this->createVariableSymbolContext($leftOperand);
+    }
+
+    protected function createPropertySymbolContext(
+        MemberAccessExpression $leftOperand,
+        Frame $frame
+    ): SymbolContext {
+        assert($leftOperand->dereferencableExpression instanceof Variable);
+
+        $classVariableName = $leftOperand->dereferencableExpression->getName();
+
+        try {
+            $classType = $frame->locals()->byName($classVariableName)->first()
+                ->symbolContext()->types()->best()
+            ;
+        } catch (\RuntimeException $exception) {
```

Comment:
We should add something like `Phpactor\WorseReflection\Core\Inference\Exception\VariableNotFound` exception and replacing all the `RuntimeException` in that collection... not necessarily in this PR.

Reply:
I might try to find the time to do it, otherwise we will never get it done.
I'm thinking about make the new exception extends `\RuntimeException` just in case I would miss any spot and then trying to identify obvious places where we used the `\RunetimeException` for this case.